modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,407 @@
+			 */
+			_prime_post_caches( $attachment_ids, false, true );
+		}
+
+		// Iterate through the matches in order of occurrence as it is relevant for whether or not to lazy-load.
+		foreach ( $matches as $match ) {
+			// Filter an image match.
+			if ( isset( $images[ $match[0] ] ) ) {
+				$filtered_image = $match[0];
+				$attachment_id  = $images[ $match[0] ];
+				/**
+				 * Filters img tag that will be injected into the content.
+				 *
+				 * @param string $filtered_image the img tag with attributes being created that will
+				 *                                    replace the source img tag in the content.
+				 * @param string $context Optional. Additional context to pass to the filters.
+				 *                        Defaults to `current_filter()` when not set.
+				 * @param int $attachment_id the ID of the image attachment.
+				 *
+				 * @since 1.0.0
+				 */
+				$filtered_image = apply_filters( 'wp_dominant_color_img_tag_add_adjust', $filtered_image, $context, $attachment_id );
+
+				if ( $filtered_image !== $match[0] ) {
+					$content = str_replace( $match[0], $filtered_image, $content );
+				}
+			}
+		}
+
+		return $content;
+	}
+
+	public function set_wp_image_editors() {
+
+		require_once 'WP_Image_Editor_GD_With_Color.php';
+		require_once 'WP_Image_Editor_Imagick_With_Color.php';
+
+		return array( 'WP_Image_Editor_GD_With_Color', 'WP_Image_Editor_Imagick_With_Color' );
+	}
+
+	/**
+	 * Get dominant color of image
+	 *
+	 * @param integer $id
+	 * @param string  $default_color
+	 *
+	 * @return string
+	 */
+	public function get_dominant_color( $id, $default_color = 'eee' ) {
+
+		add_filter( 'wp_image_editors', array( $this, 'set_wp_image_editors' ) );
+
+		$file = get_attached_file( $id );
+
+		$dominant_color = wp_get_image_editor( $file )->get_dominant_color( $default_color );
+
+		remove_filter( 'wp_image_editors', array( $this, 'set_wp_image_editors' ) );
+
+		return $dominant_color;
+
+	}
+
+
+	/**
+	 * Works out if color has transparency
+	 *
+	 * @param integer $id
+	 *
+	 * @return bool
+	 */
+	public function get_has_transparency( $id ) {
+
+		add_filter( 'wp_image_editors', array( $this, 'set_wp_image_editors' ) );
+
+		$file = get_attached_file( $id );
+		if ( wp_get_image_mime( $file ) === 'image/webp' ) {
+			$webpinfo = $this->webpinfo( $file );
+
+			return $webpinfo['Alpha'];
+		}
+
+		$has_transparency = wp_get_image_editor( $file )->get_has_transparency();
+
+		remove_filter( 'wp_image_editors', array( $this, 'set_wp_image_editors' ) );
+
+		return $has_transparency;
+	}
+
+	/**
+	 * Get WebP file info.
+	 *
+	 * @link https://www.php.net/manual/en/function.pack.php unpack format reference.
+	 * @link https://developers.google.com/speed/webp/docs/riff_container WebP document.
+	 *
+	 * @param string $file
+	 *
+	 * @return array|false Return associative array if success, return `false` for otherwise.
+	 */
+	private function webpinfo( $file ) {
```

Comment:
How does this with wp_get_webp_info in core? Can we use that?

Reply:
this is just looking for alpha channel
renamed to make clear